build-scripts/docker-images.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+set -ex
+registry=${2:-localhost:5001} # TODO make public cfengine AWS ECR registry :)
+
+cat <<EOF >Dockerfile-ubuntu-20-systemd
+FROM ubuntu:20.04
+RUN apt-get update && apt-get install -q -y systemd wget sudo
+CMD [ "/lib/systemd/systemd" ]
+EOF
+
+for name in cfengine-hub cfengine-agent; do
+  if [ "$name" = "cfengine-agent" ]; then
+    install_type="agent"
+  else
+    install_type="hub"
+  fi
+  if docker ps -a | grep $name; then
+    docker ps -a | grep $name | awk '{print $1}' | xargs docker stop
+    docker ps -a | grep $name | awk '{print $1}' | xargs docker rm
+    docker rm $name
+  fi
```

Comment:
i wonder if you can use
```
docker ps -a --quiet --filter name=$name
```
to get container SHAs, and then pipe them to
```
xargs --no-run-if-empty docker stop
```

Actually, given that `docker stop` prints back SHA of stopped container, could you pipe it to `xargs --no-run-if-empty docker stop`?

```suggestion
  docker ps -a --quiet --filter name=$name | xargs --no-run-if-empty docker stop | xargs --no-run-if-empty docker rm -f
```

Reply:
I tend to lean against more arguments and try to keep things "simple" but it is a lovely suggestion.

Reply:
yep, that's a valid point: we need to keep a proper balance between "simple" code which is easy to understand by a human, and a code which fits into one line but takes more time to read (by human) and understand. :balance_scale: